components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -8000,44 +8002,91 @@ public Comment getComment(ApiTypeWrapper apiTypeWrapper, String commentId, Integ
-                    comment.setUser(resultSet.getString("CREATED_BY"));
-                    comment.setCreatedTime(resultSet.getTimestamp("CREATED_TIME"));
-                    comment.setUpdatedTime(resultSet.getTimestamp("UPDATED_TIME"));
-                    comment.setApiId(resultSet.getString("API_ID"));
-                    comment.setParentCommentID(resultSet.getString("PARENT_COMMENT_ID"));
-                    comment.setEntryPoint(resultSet.getString("ENTRY_POINT"));
-                    comment.setCategory(resultSet.getString("CATEGORY"));
-                    comment.setReplies(Arrays.asList(getComments(identifier, resultSet.getString("COMMENT_ID"),
-                            connection)));
-                    commentList.add(comment);
+        String sqlQueryForCount;
+        if (parentCommentID == null){
+            sqlQueryForCount  = SQLConstants.GET_ROOT_COMMENTS_COUNT_SQL;
+        }else {
+            sqlQueryForCount  = SQLConstants.GET_REPLIES_COUNT_SQL;
+        }
+        try (PreparedStatement prepStmtForCount = connection.prepareStatement(sqlQueryForCount)) {
+            prepStmtForCount.setString(1, APIUtil.replaceEmailDomainBack(identifier.getProviderName()));
+            prepStmtForCount.setString(2, identifier.getName());
+            prepStmtForCount.setString(3, identifier.getVersion());
+            if (parentCommentID != null){
+                prepStmtForCount.setString(4, parentCommentID);
+            }
+            try (ResultSet resultSetForCount = prepStmtForCount.executeQuery()) {
+                while (resultSetForCount.next()) {
+                    total = resultSetForCount.getInt("COMMENT_COUNT");
+                }
+                if (total > 0) {
+                    if (parentCommentID == null) {
+                        sqlQuery = SQLConstants.GET_ROOT_COMMENTS_SQL;
+                    } else {
+                        sqlQuery = SQLConstants.GET_REPLIES_SQL;
+                    }
+                    try (PreparedStatement prepStmt = connection.prepareStatement(sqlQuery)) {
+                        prepStmt.setString(1, APIUtil.replaceEmailDomainBack(identifier.getProviderName()));
+                        prepStmt.setString(2, identifier.getName());
+                        prepStmt.setString(3, identifier.getVersion());
+                        if (parentCommentID != null) {
+                            prepStmt.setString(4, parentCommentID);
+                            // temporary commented below two lines as mssql doesn't support limit and offset
+//                            prepStmt.setInt(5, limit);
+//                            prepStmt.setInt(6, offset);
+                        } else {
+                            // temporary commented below two lines as mssql doesn't support limit and offset
+//                            prepStmt.setInt(4, limit);
+//                            prepStmt.setInt(5, offset);
```

Comment:
shall we create an issue for this and address separately?

Reply:
This was solved